app/assets/javascripts/hyrax/admin/collection_type/settings.es6

```diff
@@ -0,0 +1,55 @@
+// Enable/disable sharing APPLIES_TO_NEW_WORKS checkbox based on the state of checkbox SHARABLE
+export default class {
+  constructor(element) {
+    this.element = element
+  }
+
+  setup() {
+    // Watch for changes to "sharable" checkbox
+    $("#collection_type_sharable").on('change', () => { this.sharableChanged() })
```

Comment:
I'd prefer we do all the sector binding in one spot rather than have every method know about these selectors.  e.g.

in setup do:
```
this.sharable_checkbox = $("#collection_type_sharable")
this.applies_to_new_works = $("#collection_type_share_applies_to_new_works")
this.container = $("#sharable-applies-to-new-works-setting-checkbox-container")
this.label = $("#sharable-applies-to-new-works-setting-label")
```